app/components/search.tsx

```diff
@@ -1,36 +1,69 @@
-import {useState, useEffect, useRef, MutableRefObject, FocusEvent} from 'react'
+import {useState, useEffect, useCallback, useRef, MutableRefObject, FocusEvent} from 'react'
 import debounce from 'lodash/debounce'
+import {Searcher, Question as QuestionType, SearchResult} from 'stampy-search'
 import {AddQuestion} from '~/routes/questions/add'
 import {Action, ActionType} from '~/routes/questions/actions'
 import {MagnifyingGlass, Edit} from '~/components/icons-generated'
-import {useSearch, Question as QuestionType, SearchResult} from '~/hooks/search'
 import AutoHeight from 'react-auto-height'
 import Dialog from '~/components/dialog'
 
 type Props = {
   onSiteAnswersRef: MutableRefObject<QuestionType[]>
+  initialQuery?: string
   openQuestionTitles: string[]
   onSelect: (pageid: string, title: string) => void
 }
 
 const empty: [] = []
 
-export default function Search({onSiteAnswersRef, openQuestionTitles, onSelect}: Props) {
-  const [showResults, setShowResults] = useState(false)
+export default function Search({
+  onSiteAnswersRef,
+  initialQuery,
+  openQuestionTitles,
+  onSelect,
+}: Props) {
+  const [showResults, setShowResults] = useState(initialQuery !== undefined)
   const [showMore, setShowMore] = useState(false)
   const searchInputRef = useRef('')
 
-  const {search, arePendingSearches, results} = useSearch(onSiteAnswersRef)
+  const [arePendingSearches, setPendingSearches] = useState(false)
+  const [results, setResults] = useState([] as SearchResult[])
+  const [searcher, setSearcher] = useState<Searcher>()
+
+  useEffect(() => {
+    setSearcher(
+      new Searcher({
+        getAllQuestions: () => onSiteAnswersRef.current,
+        onResolveCallback: (query?: string, res?: SearchResult[] | null) => {
+          if (res) {
+            setPendingSearches(false)
+            setResults(res)
+          }
+        },
+      })
+    )
+  }, [onSiteAnswersRef])
 
-  const searchFn = (rawValue: string) => {
-    const value = rawValue.trim()
-    if (value === searchInputRef.current) return
+  const searchFn = useCallback(
+    (rawValue: string) => {
+      const value = rawValue.trim()
+      if (value === searchInputRef.current) return
 
-    searchInputRef.current = value
+      searchInputRef.current = value
 
-    search(value)
-    logSearch(value)
-  }
+      setPendingSearches(true)
+      searcher && searcher.searchLive(value)
```

Comment:
this is probably mostly for TS reasons, but it shouldn't be a valid option to NOT perform the search at this step, if it "always works" it should throw an exception after future changes that make it suddenly "not work"